rand_chacha/src/chacha.rs

```diff
@@ -145,35 +89,55 @@ macro_rules! chacha_impl {
         ///       http://www.ecrypt.eu.org/stream/)
         #[derive(Clone, Debug)]
         pub struct $ChaChaXRng {
-            rng: BlockRng<$ChaChaXCore>,
+            core: $ChaChaXCore,
+            buffer: [u32; 64],
+        }
+
+        impl $ChaChaXRng {
+            fn buffer_index(&self) -> u32 {
+                self.buffer[0]
+            }
+
+            fn generate_and_set(&mut self, index: usize) {
+                assert!(index < self.buffer.len());
+                self.buffer[0] = if index != 0 {
+                    self.core.next_block(&mut self.buffer);
+                    index as u32
+                } else {
+                    self.buffer.len() as u32
+                }
+            }
         }
```

Comment:
These `fns` are needed to support get/set word-pos fns. They were provided by [`BlockRng`](https://docs.rs/rand_core/latest/rand_core/block/struct.BlockRng.html).